roles/openstack/nova/tasks/cleanup.yml

```diff
@@ -0,0 +1,45 @@
+---
+# This file is used to cleanup OpenStack nova resources
+
+- name: Register nova servers details
+  os_server_facts:
+      cloud: "{{ provisioner.cloud | default(omit) }}"
+      server: "{{ prefix }}*"
+  register: servers_details
+- debug: var=servers_details
+
+- name: Set floating IP address
+  set_fact:
+      floating_ip_address: "{{ item.value[0].interface_ip }}"
+  with_dict: servers_details.ansible_facts
+  ignore_errors: yes
```

Comment:
I meant something else entirely, but after your explanation I understand why it's implemented this way and I am fine with it, so just proposing name change to be more meaningful.